doc/RecordingControl.xml

```diff
@@ -1684,8 +1766,220 @@ SetRecordingJobMode(JobToken, Active)
+          </imageobject>
+        </mediaobject>
+      </figure>
+      <para>
+        Each segment is uploaded as one object to the object storage.
+        This annex also defines other object types to help the receiver process the data.
+      </para>
+    </section>
+    <section>
+      <title>Spans</title>
+      <para>
+        When an underlying recording is started, the device shall start a new span.
+        Inversely, when the underlying recording is stopped, the device shall close the current span of the underlying recording.
+        When a <literal>SpanDuration</literal> is configured, the underlying recording is active, and the duration of the current span reaches the configured <literal>SpanDuration</literal>, the device shall close the current span of the underlying recording and start a new span.
+      </para>
+      <para>
+        All fragments of a span shall be independently processable and presentable with the information of a <literal>MovieBox</literal> in any of the segments of the span.
+        If the information in the <literal>MovieBox</literal> required to process the fragments needs to change while the underlying recording is active, the device shall close the current span of the underlying recording and start a new span.
+      </para>
+    </section>
+    <section>
+      <title>Segments</title>
+      <para>
+        The device shall start new segments when a span is started and close all current segments of the underlying span when it is closed.
+        The device shall close the current segments of the underlying span and start new segments when the duration of the current segments of the underlying span reaches the configured <literal>SegmentDuration</literal>.
+      </para>
+      <para>
+        The device shall generate each segment as a fragmented file according to ISO/IEC 14496-12, the ISO base media file format.
+        The recording formats provide specification about segment types, their contents, and restrictions on the ISO base media file format in the following sections.
+      </para>
+    </section>
+    <section xml:id="_refCloudRecordingSegmentObjects">
+      <title>Segment objects</title>
+      <para>
+        The device shall upload each segment as one object to the object storage.
+        Each object and corresponding segment is identified by its object key.
+        The device shall generate the object key for a segment according to the following ABNF <literal>key</literal> rule (according to RFC 5234):
+      </para>
+      <programlisting>key      = lead "/" time "." counter "." extension
+lead     = [prefix "/"] date ["/" postfix]
+counter  = 1*DIGIT
+date     = fullyear "-" month "-" mday
+time     = hour "-" minute "-" second [secfrac] "Z"
+fullyear = 4DIGIT
+month    = 2DIGIT ; 01-12
+mday     = 2DIGIT ; 01-31
+hour     = 2DIGIT ; 00-23
+minute   = 2DIGIT ; 00-59
+second   = 2DIGIT ; 00-60 (due to leap second rules)
+secfrac  = "." 1*DIGIT</programlisting>
```

Comment:
Parsing that part might be tricky if there is no limit to the precision (number of digits).
6 digits (or 1 microsecond) might be an acceptable limit?

Reply:
I would suggest changing this to a fixed `3DIGIT` representation.

Reply:
As discussed, `secfrac` is now limited to 6 digits.